hrana/lib/src/rpc_http_client.dart

```diff
@@ -0,0 +1,267 @@
+import 'dart:async';
+import 'dart:developer';
+
+import 'package:hrana/src/exception.dart';
+import 'package:hrana/src/protocol.pb.dart' as proto;
+import 'package:hrana/src/rpc_client.dart';
+import 'package:http/http.dart' as http;
+
+final class HranaHttpClient implements HranaClient {
+  HranaHttpClient._(
+    this._uri, {
+    String? jwtToken,
+    http.Client? httpClient,
+  }) : _httpClient = _AuthorizingHttpClient(
+          inner: httpClient,
+          jwtToken: jwtToken,
+        );
+
+  final Uri _uri;
+  final http.Client _httpClient;
+  final Completer<void> _closed = Completer();
+
+  var _nextStreamId = 0;
+  var _nextSqlTextId = 0;
+  final Map<SqlStreamId, _SqlStreamInfo> _streams = {};
+
+  static void _log(String message) {
+    log(
+      message,
+      name: 'Hrana',
+      level: 300, // Level.FINEST
+    );
+  }
+
+  static Future<HranaHttpClient> connect(
+    Uri uri, {
+    String? jwtToken,
+    http.Client? httpClient,
+  }) async {
+    final client = HranaHttpClient._(
+      uri,
+      jwtToken: jwtToken,
+      httpClient: httpClient,
+    );
+    await client._ensureProtocolSupport();
+    return client;
+  }
+
+  /// Ensures that the server supports the Hrana V3 Protobuf protocol.
+  Future<void> _ensureProtocolSupport() async {
+    final url = _uri.resolve('./v3-protobuf');
+    final response = await _httpClient.get(url);
+    if (response.statusCode >= 300) {
```

Comment:
Technically I think we should also reject 1xx status codes? So maybe

```suggestion
    if (response.statusCode case < 200 || >= 300) {
```

Reply:
Good point 👍